framework/src/main/java/org/checkerframework/common/value/ValueTreeAnnotator.java

```diff
@@ -398,6 +399,20 @@ public Void visitMethodInvocation(MethodInvocationTree tree, AnnotatedTypeMirror
             }
         }
 
+        if (atypeFactory
+                .getMethodIdentifier()
+                .isArraysCopyOfInvocation(tree, atypeFactory.getProcessingEnv())) {
+            List<? extends ExpressionTree> args = tree.getArguments();
+            if (args.size() != 2) {
+                throw new BugInCF(
+                        "Arrays.copyOf() should have 2 arguments. This point should not have reached");
+            }
+            Range range =
```

Comment:
`range` can be null here, but `createArrayLenRange` assumes its input in non-null. Can you add a null check and only do the `type.replaceAnnotation(atypeFactory.createArrayLenRangeAnnotation(range));` if the range is non-null?